play/android-publisher/src/main/kotlin/com/github/triplet/gradle/androidpublisher/internal/AndroidPublisher.kt

```diff
@@ -42,6 +43,30 @@ internal fun createPublisher(credentials: InputStream): AndroidPublisher {
     ).setApplicationName(PLUGIN_NAME).build()
 }
 
+internal fun createPublisher(impersonateServiceAccount: String?): AndroidPublisher {
+    val transport = buildTransport()
+
+    val appDefaultCreds = GoogleCredentials.getApplicationDefault()
+
+    val credential = if (impersonateServiceAccount != null) {
+        ImpersonatedCredentials.newBuilder()
+                .setSourceCredentials(appDefaultCreds)
+                .setTargetPrincipal(impersonateServiceAccount)
+                .setScopes(listOf(AndroidPublisherScopes.ANDROIDPUBLISHER))
+                .setLifetime(300)
```

Comment:
Max this out to 3600. We don't want folks with really long builds getting timed out.

Reply:
It seems the tokens are generated as the task is about to make a HTTP call (rather than at the start of a build) and also automatically refreshed if "stale" (i.e. expired or within 5mins of expiry), so the thinking was that this lower value was pretty safe.. 
But an hour isn't the end of the world so happy to just max it out if preferred...

Reply:
Yeah, I don't see how a token living for an hour is going to hurt much more than 5 mins.